sql/src/test/java/org/apache/druid/sql/calcite/util/QueryFramework.java

```diff
@@ -0,0 +1,439 @@
+import org.apache.druid.sql.calcite.planner.DruidOperatorTable;
+import org.apache.druid.sql.calcite.planner.PlannerConfig;
+import org.apache.druid.sql.calcite.planner.PlannerFactory;
+import org.apache.druid.sql.calcite.run.NativeSqlEngine;
+import org.apache.druid.sql.calcite.run.SqlEngine;
+import org.apache.druid.sql.calcite.schema.DruidSchemaCatalog;
+import org.apache.druid.sql.calcite.schema.NoopDruidSchemaManager;
+import org.apache.druid.sql.calcite.view.DruidViewMacroFactory;
+import org.apache.druid.sql.calcite.view.InProcessViewManager;
+import org.apache.druid.timeline.DataSegment;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Builds the infrastructure needed to run Calcite tests. Building splits into
+ * two parts: a constant part and a per-test part. The constant part includes
+ * the congolmerate and walker (that is, the implementation of the test data),
+ * while the per-query part includes the schema, planner factory and SQL
+ * statement factory. The key reason for the split is that the per-test part
+ * depends on the value of the {@link PlannerConfig} object, which varies between
+ * tests.
+ * <p>
+ * The builder accepts the injector to use. "Calcite tests" use the injector
+ * defined in {@link CalciteTests#INJECTOR}, while other tests can pass in the
+ * preferred injector.
+ * <p>
+ * The framework allows the test to customize many of the framework components.
+ * Since those components tend to depend on other framework components, we use
+ * an indirection, {@link QueryFramework.QueryComponentSupplier QueryComponentSupplier}
+ * to build those components. The methods of this interface match the methods
+ * in @link org.apache.druid.sql.calcite.BaseCalciteQueryTest BaseCalciteQueryTest}
+ * so that those tests can customize the framework just by overriding methods.
+ * (This was done to maintain backward compatibility with earlier versions of the
+ * code.) Other tests can implement {@code QueryComponentSupplier} directly by
+ * extending {@link QueryFramework.StandardComponentSupplier StandardComponentSupplier}.
+ * <p>
+ * The framework should be built once per test class (not once per test method.)
+ * Then, per method, call {@link #statementFactory(PlannerConfig, AuthConfig)} to
+ * obtain a the test-specific planner and wrapper classes for that test. After
+ * that, tests use the various SQL statement classes to run tests. For tests
+ * based on {@code BaseCalciteQueryTest}, the statements are wrapped by the
+ * various {@code testQuery()} methods.
+ * <p>
+ * The framework holds on to the framework components. You can obtain the injector,
+ * object mapper and other items by calling the various methods. The objects
+ * are those created by the provided injector, or in this class, using objects
+ * from that injector.
+ */
+public class QueryFramework
```

Comment:
nit: should we call this `QueryTestFramework` or `SqlTestFramework` or something?